src/components/VisitorInfo/VisitorInfo.jsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import './VisitorInfo.css';
+
+const VisitorInfo = ({ favorites }) => (
+  <div className={'visitor-info'}>
+    {
+      Object.keys(favorites).length === 0 ? null :
+      <span>
+        <h2>{'Your Favorites'}</h2>
+        <div className={'border-magic'}>
+        {
+          Object.keys(favorites).map((item, index) => (
+            <div className={'favorite-row'} key={`favorite-${index}`}>
```

Comment:
I haven't read the links, but would something like using the favoriteType and the index as the key be better?

Reply:
They recommend using an ID attribute if it's available. The index works if there isn't a unique ID around though.

Reply:
Yeah, I guess I had a bad experience in the past with using too many IDs so I avoid them :)

Reply:
Ok, ids in the data.  (I finally read the links)  The second link shows it using the data itself.  Ex, key=number.toString().  I get it about the re-rendering stuff.  Since the favoriteType is unique, I was thinking that would be the best option???